File: GeneratedEntityReaderClasses/Components/PhysicsRagdollComponent.cs
namespace GeneratedNoitaClasses;
public class PhysicsRagdollComponent : NoitaComponentBase,  iNoitaType<PhysicsRagdollComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaString filename { get; set; } = new();
	public NoitaString filenames { get; set; } = new();
	public NoitaFloat offset_x { get; set; } = new();
	public NoitaFloat offset_y { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		filename.Read(s);
		filenames.Read(s);
		offset_x.Read(s);
		offset_y.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		filename.Write(s);
		filenames.Write(s);
		offset_x.Write(s);
		offset_y.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/PhysicsRagdollComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class PhysicsRagdollComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    filename:NoitaString = field(default_factory=NoitaString)
    filenames:NoitaString = field(default_factory=NoitaString)
    offset_x:NoitaFloat = field(default_factory=NoitaFloat)
    offset_y:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.filename.Read(s)
        self.filenames.Read(s)
        self.offset_x.Read(s)
        self.offset_y.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.filename.Write(s)
        self.filenames.Write(s)
        self.offset_x.Write(s)
        self.offset_y.Write(s)

